integration-cli/docker_cli_swarm_test.go

```diff
@@ -194,3 +194,19 @@ func (s *DockerSwarmSuite) TestSwarmNodeTaskListFilter(c *check.C) {
 	c.Assert(out, checker.Not(checker.Contains), name+".2")
 	c.Assert(out, checker.Not(checker.Contains), name+".3")
 }
+
+// Test case for #25375
+func (s *DockerSwarmSuite) TestSwarmPublishAdd(c *check.C) {
```

Comment:
I think this can be tested with a unit test, we don't need the extra cost of an integration test to fully test this behaviour.